specs/language/expressions.tex

```diff
@@ -161,8 +161,10 @@
 
 \p A \textit{postfix-expression} followed by an expression in square brackets
 (\texttt{[ ]}) is a subscript expression. In an array subscript expression of
-the form \texttt{E1[E2]}, \texttt{E1} must either be a variable of array of
-\texttt{T[]}, or an object of type \texttt{T} where \texttt{T} provides an
+the form \texttt{E1[E2]}, \texttt{E1} must either be a variable of array,
+vector, matrix, typed buffer (\ref{Resources.tybuf}), or structured buffer
+(\ref{Resources.stbufs}), with elements of type \texttt{T[]} or an
```

Comment:
Resource types have overloads for `operator[]` even in DXC. The original language did miss vector and matrix types, but the addition of resources is redundant.